lib/index.js

```diff
@@ -119,8 +122,24 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
           });
       }
       const {account: accountId, email} = req.query;
-      // if a token does not exist `brAuthnToken.get` throws a NotFoundError
-      const token = await brAuthnToken.get({accountId, email, type});
+      // if a token does not exist `brAuthnToken.get`
+      // returns fake hash parameters.
+      let token;
+      try {
+        token = await brAuthnToken.get({accountId, email, type});
+      } catch(e) {
+        const salt = await getRandomValues(new Uint8Array(16));
+        const b64Salt = _toBase64NoPad(salt);
+        token = {
+          hashParameters: {
+            id: 'pbkdf2-sha512',
+            params: {i: 100000},
+            salt: b64Salt
+          }
+        };
+      }
+      const delay = cfg.delay.mean + Math.random() * cfg.delay.jitter;
+      await _sleep({delay});
```

Comment:
So this approach will reveal a different salt every time for the same username. That will reveal that the username is not in the system. It will also use up entropy for each request, but that's probably not a significant concern anymore these days.

I think what we want to do instead is set a not-so-secret hmac key in the config that can be overwritten in production. Then we should hmac the given username / email (whatever the input is) and use that to produce a consistent salt to use in what gets returned. We should do that on every request -- concurrently with getting the token from the database. This should help minimizing timing differences.

We should then return either the real token (if found) or the auto-generated token if not found. That should do a better job with hiding the information we're trying to hide here.